packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/lib/LinkQueueLogger.ts

```diff
@@ -0,0 +1,209 @@
+import type { ILinkQueue, ILink } from '@comunica/bus-rdf-resolve-hypermedia-links-queue';
+import { LinkQueueWrapper } from '@comunica/bus-rdf-resolve-hypermedia-links-queue';
+import type { Logger } from '@comunica/types';
+import { PRODUCED_BY_ACTOR } from '@comunica/types-link-traversal';
+import type { Algebra } from 'sparqlalgebrajs';
+
+/**
+ * A link queue that log information about event happening in the queue
+ */
+export class LinkQueueLogger extends LinkQueueWrapper {
+  public readonly query: Algebra.Operation;
+  private readonly logger: Logger;
+  private readonly reachabilityRatio: IReachabilityRatio = {
+    pushEvent: {},
+    popEvent: {},
+  };
+
+  public static readonly LINK_QUEUE_EVENT_NAME = '<Link queue occupancy>';
+  public static readonly LINK_QUEUE_DIDNT_START_EMPTY_MESSAGE = 'the link queue didn\'t start empty';
+
+  /**
+   *
+   * @param {ILinkQueue & IOptionalLinkQueueParameters} linkQueue - The link queue with optional parameters
+   * @param {Algebra.Operation} query - The current query
+   * @param {Logger} logger - The logger where the events are output
+   */
+  public constructor(linkQueue: ILinkQueue & IOptionalLinkQueueParameters, query: Algebra.Operation, logger: Logger) {
+    super(linkQueue);
+    this.logger = logger;
+
+    if (!linkQueue.isEmpty()) {
```

Comment:
I just wanted to be save since technically nothing stop this from happening.

Reply:
And the results then would not be accurate.

Reply:
> since technically nothing stop this from happening.

When can this occur then?

Reply:
It is because the wrapper is not responsible to create the queue so technically anyone could wrapped the queue at any moment and start adding things into it. Of course in the current code it doesn't happen but if it were to happen the class would not respect its contract.

Reply:
> Of course in the current code it doesn't happen but if it were to happen the class would not respect its contract.

Let's remove it then. Unnecessary checks unnecessarily slow down the code.